src/main/java/carpetextra/dispenser/behaviors/PlaceBoatOnIceDispenserBehavior.java

```diff
@@ -0,0 +1,60 @@
+package carpetextra.dispenser.behaviors;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.DispenserBlock;
+import net.minecraft.block.dispenser.FallibleItemDispenserBehavior;
+import net.minecraft.entity.vehicle.BoatEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.tag.BlockTags;
+import net.minecraft.util.math.BlockPointer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+
+import java.util.Map;
+
+public class PlaceBoatOnIceDispenserBehavior extends FallibleItemDispenserBehavior {
+
+    private static final Map<Item, BoatEntity.Type> BOAT_ITEM_TO_TYPE = Map.of(
+            Items.SPRUCE_BOAT, BoatEntity.Type.SPRUCE,
```

Comment:
Is there a smarter way to achieve this? BoatItem.type is private, so I don't know if there's a way to obtain it directly.

Reply:
You could do `boatEntity.setBoatType(BoatEntity.Type.getType(item.getName().asString()));`, which also accounts for modded boat types.

Reply:
I like this idea a lot, but it doesn't seem to work quite yet:
![image](https://user-images.githubusercontent.com/13875418/148704079-86af28f1-be50-4dad-93e0-33bbe0e362d2.png)
Seems that the `getName().asString()` just return nothing.
Looks like the only field I can get is translation key, which ends up like this:
![image](https://user-images.githubusercontent.com/13875418/148705936-661cb95d-f8fc-4fba-9776-db32b855c774.png)
Would this be preferable to the above? If so I'll commit it

Reply:
Nevermind, I figured out a better solution! TIL mixins allow you to get private fields